src/server/chromium/crNetworkManager.ts

```diff
@@ -337,10 +342,16 @@ export class CRNetworkManager {
     if (request._interceptionId)
       this._attemptedAuthentications.delete(request._interceptionId);
     this._page._frameManager.requestReceivedResponse(response);
-    this._page._frameManager.requestFinished(request.request);
+    this._page._frameManager.requestFinished(request.request, response);
+  }
+
+  _onResponseReceivedExtraInfo(event: Protocol.Network.responseReceivedExtraInfoPayload) {
+    this._responseExtraInfoTracker.responseReceivedExtraInfo(event);
   }
 
   _onResponseReceived(event: Protocol.Network.responseReceivedPayload) {
+    this._responseExtraInfoTracker.responseReceived(event);
```

Comment:
I am a bit confused that we notify ResponseExtraInfoTracker here, and also in _createResponse called 6 lines below.

Reply:
They are sending two distinct signals to tracker. As you can see, tracker repeats the CDP api. This is to make sure that it is easy to read it - CDP has well-understood contract. So our tracker is an engine that consumes two inputs: CDP on one side and created Playwright Responses on the other side.